vscode-sbcl-clpm.md

```diff
@@ -0,0 +1,472 @@
+  sharing dependencies across your entire machine, and things like
+  [quicklisp](https://www.quicklisp.org/beta/) go along with that.
+- This run-through uses [CLPM](https://clpm.dev). CLPM is more like the
+  project managers that you are used to in other languages. I want the
+  ability to pin versions in my projects, source projects that only
+  exist in Github, not share dependencies across my entire computer, and
+  use libraries in quicklisp. For that, and [other
+  reasons](https://www.clpm.dev/#project-goals), I use clpm. There are
+  other tools, and at some point, I will write about them, but today
+  they don't matter. You don't have to do things my way, but you will
+  have a better time if you do for this example.
+- I use rlwrap, which gives me history and search in my repl. If you
+  don't have access to it, leave it off of any example commands that I
+  provide.
+
+#### Steps to create a new project
+1. Create a folder for your new project
+   1. Start a repl
+      ```bash
+      rlwrap sbcl
+      ```
+   2. Create a directory for the project called `experiment`
+      ~~~lisp
+      (ensure-directories-exist "experiment/") ;; That last slash matters
+      ~~~
+   3. Exit the repl
+      ~~~lisp
+      (sb-ext:exit) ;; You can also just do a ctrl-d
+      ~~~
+2. Open vscode in your newly created directory
+   ```bash
+   cd experiment
+   code .
+   ```
+3. Create new Common Lisp System using Alive.
+
+   _Note: You don't have to use Alive to create a skeleton project.  There are
+   dozens of ways to do this, I am just trying to keep it simple, so do this
+   however you want._
+
+   1. Inside of vscode, Open Command Palette on the menu at the top
+      `View/Command Palette`
+   2. Generate skeleton: `Alive: System Skeleton`
+   3. The previous command should have generated the following directory
+      structure
+      - src
+        - app.lisp
+           ~~~lisp
+           (defpackage :app
+             (:use :cl))
+
+           (in-package :app)
+           ~~~
+      - test
+        - all.lisp
+           ~~~lisp
+           (defpackage :test/all
+             (:use :cl
+                   :app)
+             (:export :test-suite))
+
+           (in-package :test/all)
+
+           (defun test-suite ()
+             (format T "Test Suite~%"))
+           ~~~
+      - experiment.asd
+         ~~~lisp
+         (in-package :asdf-user)
+
+         (defsystem "experiment"
+           :class :package-inferred-system
+           :depends-on ("experiment/src/app")
+           :description ""
+           :in-order-to ((test-op (load-op "experiment/test/all")))
+           :perform (test-op (o c) (symbol-call :test/all :test-suite)))
+
+         (defsystem "experiment/test"
+           :depends-on ("experiment/test/all"))
+
+         (register-system-packages "experiment/src/app" '(:app))
+         (register-system-packages "experiment/test/all" '(:test/all))
+         ~~~
```

Comment:
so far it's the same of the TLDR; version above, isn't it?

Reply:
Yea, this whole section is the same, but with explanations for each step.  Maybe that is unnecessary?  My rational was that some people just want to stampede to the end so they can see it working, and others want to have an idea of what is happening from the beginning.  That might have been a poor assumption.

Reply:
I just nuked the TL;DR; and trimmed down the other.